src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2472,31 +2470,16 @@ internal override void GetDiagnostics(CompilationStage stage, bool includeEarlie
                 var syntaxTrees = this.SyntaxTrees;
                 if (this.Options.ConcurrentBuild)
                 {
-                    var parallelOptions = cancellationToken.CanBeCanceled
-                                        ? new ParallelOptions() { CancellationToken = cancellationToken }
-                                        : DefaultParallelOptions;
-
-                    Parallel.For(0, syntaxTrees.Length, parallelOptions,
+                    RoslynParallel.For(
+                        0,
+                        syntaxTrees.Length,
                         UICultureUtilities.WithCurrentUICulture<int>(i =>
```

Comment:
nit: we never really need the index, or hte 0->length.  Consider a `RoslynParallel.ForEach` that just operates on a collection and gives you each element to process.

Reply:
I noticed this, but then you end up with a whole new wrapper for an API only used in one scenario (two locations), plus `Parallel` is generally discouraged in most of our code so it's unlikely to have more uses in the future.

Reply:
Wfm